obvs/lenses.py

```diff
@@ -314,47 +312,55 @@ class PatchscopeLogitLens(BaseLogitLens):
 
     The source layer l and position i can vary.
 
-    In words: The logit-lens maps the hidden state at position i of layer l of the model M
-        to the last layer of that same model. It is equal to taking the hidden state and
-        applying unembed to it.
+        In words: The logit-lens maps the hidden state at position i of layer l of the model M
+            to the last layer of that same model. It is equal to taking the hidden state and
+            applying unembed to it.
     """
 
-    def run(self, substring: str, layers: list[int]):
-        """Run the logit lens for each layer in layers and each token in substring.
-
-        Args:
-            substring (str): Substring of the prompt for which the top prediction and logits
-                should be calculated.
-            layers (List[int]): Indices of Transformer Layers for which the lens should be applied
+    def __init__(self, model: str, prompt: str, device: str, layers: list[int], substring: str):
+        """
+        substring (str): Substring of the prompt for which the top prediction and logits should be calculated.
+        layers (list[int]): Indices of Transformer Layers for which the lens should be applied
         """
 
-        # get starting position and tokens of substring
+        super().__init__(model, prompt, device)
         start_pos, substring_tokens = self.patchscope.source_position_tokens(substring)
-
-        # initialize tensor for logits
+        self.start_pos = start_pos
+        self.layers = layers
+        self.substring_tokens = substring_tokens
         self.data["logits"] = torch.zeros(
             len(layers),
             len(substring_tokens),
             self.patchscope.tokenizer.vocab_size,
         )
 
+    def run(self, position: int):
```

Comment:
I get the intention, but I need to remark that this breaks the symmetry of the run methods of the two lenses.
Maybe we should change the ClassicLogitLens in the same way so that it is restored?
Or are they too different so that this does not make sense? WDYT?

Reply:
Looking over the PatchscopeLogitLens implementation again, I saw that I did something a little bit unnecessary.
The SourceContext position attribute can be a Sequence of integers (   https://github.com/obvslib/obvs/blob/main/obvs/patchscope.py#L58 ), so the loop over positions is not needed.
I think we can therefore make the run methods of both LogitLenses equal in that they process all positions at once (which should also make the PatchscopeLogitLens faster)
But this should not contaminate your PR now. Let's make a separate issue for that.